invader_payment/services/invader_payment_service.py

```diff
@@ -34,14 +35,28 @@ def _invader_find_payable_from_transaction(self, transaction):
         """
         raise NotImplementedError()
 
+    def _invader_restrict_payment_mode_ids(self):
+        return []
+
     def _invader_get_target_validator(self):
         """
         Return a cerberus validator schema fragment that specifies the
         target being paid. Implementations must extend it by populating
         the "allowed" field (eg with strings such as 'current_cart') and
         possibly adding other fields.
         """
-        return {"target": {"type": "string", "required": True, "allowed": []}}
+        validator = {
+            "target": {"type": "string", "required": True, "allowed": []},
+            "payment_mode_id": {
```

Comment:
Why????? The target MUST a value to know how to find the payable!

Reply:
Indeed your right !

Reply:
Indeed your right, I should note inject the payment_mode_id here.
I should only add this in the payment service